apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input } from "@angular/core";
+import { ThemeService } from "./theme-service";
+
+@Component({
+selector: 'app-theme-switcher',
+templateUrl: './theme-switcher.component.html',
+})
+export class ThemeExporterComponent{
+  @Input()
+  theme = ThemeService.THEMES[0];
+
+  themes = ThemeService.THEMES;
+
+  onSelect(theme: any) {
```

Comment:
Maybe `theme: ThemeService.THEMES` or string?

Reply:
Well i set it to as `onselect($event)` is passed as parameter 
i have to add `object` as well as it was throwing error that `$event` can be `string or object`  and then checking in the function if it's a `string` then the logic follows 
```onSelect(theme: string | object) {
    if(typeof theme === 'string'){
      ThemeService.setTheme(theme);
      this.theme = theme;
    }
  }

Reply:
Yes, this sounds like a better practice. `string` or `Event` type maybe?